lib/instrumentation/express-utils.js

```diff
@@ -28,12 +29,17 @@ function getStackPath (req) {
 
 // This function is also able to extract the path from a Restify request as
 // it's storing the route name on req.route.path as well
-function getPathFromRequest (req, useBase) {
+function getPathFromRequest (req, useBase, usePathAsTransactionName) {
   // Static serving route
   if (req[symbols.staticFile]) {
     return 'static file'
   }
 
+  // Enable usePathAsTransactionName config
+  if (usePathAsTransactionName) {
+    return req._parsedUrl ? req._parsedUrl.pathname : parseUrl(req).pathname
```

Comment:
If I understand the idea behind the `parseurl` module correctly, the idea is that you can just call it again with the same request and would get the same parsed object back. So you shouldn't need to access `req._parsedUrl` your self, as that's what `parseUrl(req)` does for you.

Also, `parseUrl(req)` can potentially return `undefined`, so we would need to guard against that.

This should take care of both those things:

```suggestion
    const parsed = parseUrl(req)
    return parsed && parsed.pathname
```

Reply:
Come to think of it, I'm not really sure if it's worth it using `parseurl` here.

Cons:

- It's an extra dependency
- It mutates the request object
- It only gives the user a benefit if they them selfs (or their framework) already uses the parseurl module. If not, this module doesn't give us any speed improvements

Pros:

- If the user or their framework uses the parseurl module, this will result in one allocation less than normal

@Qard any thoughts on this?

Reply:
We could just do a try/catch require of it. If the user has it in their code, we'll find it and we can check existence of the function to decide if we want to apply the logic. 🤔

Reply:
Good point @Qard. Yes I think that's the way forward. I'll make a new review